src/org/ahmedbe00/qamp/zadaci/zadaca6/task2/Robot.java

```diff
@@ -0,0 +1,79 @@
+package org.ahmedbe00.qamp.zadaci.zadaca6.task2;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Robot {
+    private int x;
+    private int y;
+    private int numberOfMoves;
+    private List<int[]> movePosition;
```

Comment:
This is not bad, and works, however think of a better idea. Instead of storing List of Arrays. Is there something better you could do.

Think about the Bookshelf.class. It stores list of Book.class objects. Could your robot do something similar, to avoid the complexity of dealing with these pairs of numbers in the form of an array.